src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -53,65 +53,104 @@ private void WriteClientMethodRequest(CodeWriter writer, LowLevelClientMethod cl
 
         private void WriteClientMethod(CodeWriter writer, LowLevelClientMethod clientMethod, bool async)
         {
-            var parameters = clientMethod.Parameters;
+            if (clientMethod.Operation.IsLongRunning)
+            {
+                WriteLongRunningOperationClientMethod(writer, clientMethod, async);
+            }
+            else
+            {
+                WriteNormalClientMethod(writer, clientMethod, async);
+            }
+        }
 
-            var responseType = async ? new CSharpType(typeof(Task<Response>)) : new CSharpType(typeof(Response));
+        private void WriteLongRunningOperationClientMethod(CodeWriter writer, LowLevelClientMethod clientMethod, bool async)
+        {
+            WriteClientMethodDecleration(writer, clientMethod, async);
 
-            writer.WriteXmlDocumentationSummary(clientMethod.Description);
+            var finalStateVia = clientMethod.Operation.LongRunningFinalStateVia switch
```

Comment:
Can we embed this logic in LongRunningFinalStateVia getter in a partial class?

Reply:
Should be able to!